ksql-engine/src/test/java/io/confluent/ksql/function/udf/math/RoundTest.java

```diff
@@ -0,0 +1,232 @@
+    assertThat(udf.round(new BigDecimal("1530000")), is(new BigDecimal("1530000")));
+    assertThat(udf.round(new BigDecimal("10.1")), is(new BigDecimal("10")));
+    assertThat(udf.round(new BigDecimal("12345.5")), is(new BigDecimal("12346")));
+    assertThat(udf.round(new BigDecimal("9.99")), is(new BigDecimal("10")));
+    assertThat(udf.round(new BigDecimal("110.1")), is(new BigDecimal("110")));
+    assertThat(udf.round(new BigDecimal("1530000.01")), is(new BigDecimal("1530000")));
+    assertThat(udf.round(new BigDecimal("9999999.99")), is(new BigDecimal("10000000")));
+  }
+
+  @Test
+  public void testRoundSimpleBigDecimalNegative() {
+    assertThat(udf.round(new BigDecimal("-1.23")), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.0")), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.5")), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1530000")), is(new BigDecimal("-1530000")));
+    assertThat(udf.round(new BigDecimal("-10.1")), is(new BigDecimal("-10")));
+    assertThat(udf.round(new BigDecimal("-12345.5")), is(new BigDecimal("-12345")));
+    assertThat(udf.round(new BigDecimal("-9.99")), is(new BigDecimal("-10")));
+    assertThat(udf.round(new BigDecimal("-110.1")), is(new BigDecimal("-110")));
+    assertThat(udf.round(new BigDecimal("-1530000.01")), is(new BigDecimal("-1530000")));
+    assertThat(udf.round(new BigDecimal("-9999999.99")), is(new BigDecimal("-10000000")));
+  }
+
+  @Test
+  public void testRoundDoubleWithDecimalPlacesPositive() {
+    assertThat(udf.round(0d, 0), is(0d));
+    assertThat(udf.round(1.0d, 0), is(1.0d));
+    assertThat(udf.round(1.1d, 0), is(1.0d));
+    assertThat(udf.round(1.5d, 0), is(2.0d));
+    assertThat(udf.round(1.75d, 0), is(2.0d));
+    assertThat(udf.round(100.1d, 0), is(100.0d));
+    assertThat(udf.round(100.5d, 0), is(101.0d));
+    assertThat(udf.round(100.75d, 0), is(101.0d));
+    assertThat(udf.round(100.10d, 1), is(100.1d));
+    assertThat(udf.round(100.11d, 1), is(100.1d));
+    assertThat(udf.round(100.15d, 1), is(100.2d));
+    assertThat(udf.round(100.17d, 1), is(100.2d));
+    assertThat(udf.round(100.110d, 2), is(100.11d));
+    assertThat(udf.round(100.111d, 2), is(100.11d));
+    assertThat(udf.round(100.115d, 2), is(100.12d));
+    assertThat(udf.round(100.117d, 2), is(100.12d));
+    assertThat(udf.round(100.1110d, 3), is(100.111d));
+    assertThat(udf.round(100.1111d, 3), is(100.111d));
+    assertThat(udf.round(100.1115d, 3), is(100.112d));
+    assertThat(udf.round(100.1117d, 3), is(100.112d));
+    assertThat(udf.round(12345.67d, -1), is(12350d));
+    assertThat(udf.round(12345.67d, -2), is(12300d));
+    assertThat(udf.round(12345.67d, -3), is(12000d));
+    assertThat(udf.round(12345.67d, -4), is(10000d));
+    assertThat(udf.round(12345.67d, -5), is(0d));
+  }
+
+  @Test
+  public void testRoundDoubleWithDecimalPlacesNegative() {
```

Comment:
nit: likewise this test is `xxxWithDecimalPlacesNegative`, but has tests with positive scale.